src/SFA.DAS.FAT.Data/Repository/TraineeshipVacancySearchRepository.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Elasticsearch.Net;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using SFA.DAS.FAT.Data.ElasticSearch;
+using SFA.DAS.FAT.Domain.Configuration;
+using SFA.DAS.FAT.Domain.Entities;
+using SFA.DAS.FAT.Domain.Interfaces;
+using SFA.DAS.FAT.Domain.Models;
+
+namespace SFA.DAS.FAT.Data.Repository
+{
+    public class TraineeshipVacancySearchRepository : IVacancySearchRepository
+    {
+        private readonly IElasticLowLevelClient _client;
+        private readonly ElasticEnvironment _environment;
+        private readonly IElasticSearchQueryBuilder _queryBuilder;
+        private readonly ILogger<TraineeshipVacancySearchRepository> _logger;
+        private const string IndexName = "-faa-traineeships";
+        
+
+        public TraineeshipVacancySearchRepository(
+            IElasticLowLevelClient client, 
+            ElasticEnvironment environment, 
+            IElasticSearchQueryBuilder queryBuilder, 
+            ILogger<TraineeshipVacancySearchRepository> logger)
+        {
+            _client = client;
+            _environment = environment;
+            _queryBuilder = queryBuilder;
+            _logger = logger;
+        }
+
+        private string TraineeshipVacanciesIndex => _environment.Prefix + IndexName;
+
+        public async Task<bool> PingAsync()
```

Comment:
Can just be `Ping` but also doesnt appear to be being used by the HealthEndpoint 🤔